hail/src/main/scala/is/hail/expr/types/Type.scala

```diff
@@ -188,6 +188,25 @@ abstract class Type extends BaseType with Serializable {
 
   def str(a: Annotation): String = if (a == null) "NA" else a.toString
 
+  final def showStr(a: Annotation, cfg: ShowStrConfig, sb: StringBuilder = null): String = {
+    val sb_ = if (sb == null)
+      new StringBuilder
+    else {
+      sb.clear()
+      sb
+    }
+    _showStrNA(a, cfg, sb_)
+    sb_.result()
+  }
+
+  final def _showStrNA(a: Annotation, cfg: ShowStrConfig, sb: StringBuilder): Unit = {
+    if (a == null)
+      sb.append(cfg.missing)
+    else
+      _showStr(a, cfg, sb)}
+
+  def _showStr(a: Annotation, cfg: ShowStrConfig, sb: StringBuilder): Unit = sb.append(str(a))
```

Comment:
I assume this does the right thing for locus, intervals, and calls because they have `toString` defined.

Reply:
yeah, the default is to just call `str`